scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/task/TaskIdImpl.java

```diff
@@ -214,4 +215,17 @@ public String toString() {
         return jobId.value() + 't' + value();
     }
 
+    /**
+     * Make a new TaskId with the given arguments.
+     *
+     * @param str the string on which to base the id.
+     * @return the new taskId
+     */
+    public static TaskId makeTaskId(String str) {
+        String[] strSplitted = str.split("t");
```

Comment:
please throw an IllegalArgumentException if str does not contain "t" or if strSplitted.length != 2

Reply:
Ok.